File: 백준/Gold/1107. 리모컨/리모컨.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
#include <cmath>

using namespace std;

int main() {
    int N, M;
    cin >> N >> M;

    vector<bool> broken_buttons(10, false);

    // 고장난 버튼 입력 받기
    for (int i = 0; i < M; i++) {
        int button;
        cin >> button;
        broken_buttons[button] = true;
    }

    // 기본값
    int min_cnt = abs(N - 100);
    int up = N, down = N;

    // 숫자 다 고장난 경우 결과는 기본값
    if (M == 10) {
        cout << min_cnt << endl;
        return 0;
    }

    auto can_press = [&](int num) {
        if (num == 0) return !broken_buttons[0];
        while (num > 0) {
            if (broken_buttons[num % 10]) return false;
            num /= 10;
        }
        return true;
        };

    // 모든 가능한 채널에 대해 확인
    for (int i = 0; i <= 1000000; i++) {
        if (can_press(i)) {
            int press = to_string(i).length() + abs(i - N);
            min_cnt = min(min_cnt, press);
        }
    }

    cout << min_cnt << endl;

    return 0;
}

File: 백준/Gold/1107. 리모컨/리모컨.py
N = int(input())
M = int(input())

# 문자형으로 숫자 받기
if M > 0:
    broken_buttons = list(input().split())

# 기본값
min_cnt = abs(N - 100)
up, down = N, N

# 숫자 다 고장난 경우 결과는 기본값
if M == 10:
    pass

# 숫자 다 멀쩡한 경우 결과
elif M == 0:
    min_cnt = min(min_cnt, len(str(N)))

else:
    cnt = 0

    while True:
        # 1씩 +, - 하면서 고장난 버튼의 숫자가 있는지 확인
        
        for broken_button in broken_buttons:
            if broken_button in str(down):
                break
        else:
            # 기본값과 채널번호의 길이 + cnt 중 작은 값 입력
            min_cnt = min(min_cnt, len(str(down)) + cnt)
            break
        
        for broken_button in broken_buttons:
            if broken_button in str(up):
                break
        else:
            # 기본값과 채널번호의 길이 + cnt 중 작은 값 입력
            min_cnt = min(min_cnt, len(str(up)) + cnt)
            break

        up += 1
        if down > 0:
            down -= 1
        cnt += 1

print(min_cnt)